tensorbay/sensor/tests/test_intrinsics.py

```diff
@@ -0,0 +1,296 @@
+
+        assert fisheye_true == 1.7543443096568176
+        assert fisheye_false == 3.0
+
+    def test_calculate_tangential_distortion(self):
+        distortioncoefficients = DistortionCoefficients(p1=1.0, p2=2.0, k1=1.0)
+        distortioncoefficients_p1 = DistortionCoefficients(p1=1.0, k1=1.0)
+
+        fisheye_true = distortioncoefficients._calculate_tangential_distortion(
+            r2=1.0, x2=2.0, y2=3.0, xy2=4.0, is_fisheye=True
+        )
+
+        with pytest.raises(AttributeError):
+            distortioncoefficients_p1_fisheye_false = (
+                distortioncoefficients_p1._calculate_tangential_distortion(
+                    r2=1.0, x2=2.0, y2=3.0, xy2=4.0, is_fisheye=False
+                )
+            )
+
+        fisheye_false = distortioncoefficients._calculate_tangential_distortion(
+            r2=1.0, x2=2.0, y2=3.0, xy2=4.0, is_fisheye=False
+        )
+
+        assert fisheye_true == (0.0, 0.0)
+        assert fisheye_false == (14.0, 15.0)
+
+    def test_list_distortions(self):
+        distortioncoefficients = DistortionCoefficients(p1=1, p2=2, k1=3, k2=4)
+        result_p = distortioncoefficients._list_distortions("p")
+
+        assert list(result_p) == [1.0, 2.0]
+
+    def test_loads(self):
+        distortioncoefficients = DistortionCoefficients.loads(_DISTORTIONCOEFFICIENTS_DATA)
+
+        assert distortioncoefficients.p1 == _DISTORTIONCOEFFICIENTS_DATA["p1"]
+        assert distortioncoefficients.p2 == _DISTORTIONCOEFFICIENTS_DATA["p2"]
+        assert distortioncoefficients.k1 == _DISTORTIONCOEFFICIENTS_DATA["k1"]
+        assert distortioncoefficients.k2 == _DISTORTIONCOEFFICIENTS_DATA["k2"]
+
+    def test_dumps(self):
+        distortioncoefficients = DistortionCoefficients(p1=1.0, p2=2.0, k1=3.0, k2=4.0)
+        assert distortioncoefficients.dumps() == _DISTORTIONCOEFFICIENTS_DATA
+
+    def test_distort(self):
+        distortioncoefficients = DistortionCoefficients(p1=1.0, p2=2.0, k1=3.0, k2=4.0)
+
+        with pytest.raises(TypeError):
+            distortioncoefficients.distort((1.0, 2.0, 3.0, 4.0), is_fisheye=True)
+
+        distored_2d_fisheye_true_1 = distortioncoefficients.distort((1.0, 2.0), is_fisheye=True)
+        distored_2d_fisheye_true_2 = distortioncoefficients.distort(
+            (1.0, 2.0, 3.0), is_fisheye=True
+        )
+
+        distored_2d_fisheye_false_1 = distortioncoefficients.distort((1.0, 2.0), is_fisheye=False)
+        distored_2d_fisheye_false_2 = distortioncoefficients.distort(
+            (1.0, 2.0, 3.0), is_fisheye=False
+        )
+
+        assert distored_2d_fisheye_true_1 == Vector2D(6.158401093771876, 12.316802187543752)
+        assert distored_2d_fisheye_true_2 == Vector2D(0.83187700005734, 1.66375400011468)
+        assert distored_2d_fisheye_false_1 == Vector2D(134.0, 253.0)
+        assert distored_2d_fisheye_false_2 == Vector2D(3.3004115226337447, 4.934156378600823)
+
+
+class TestCameraIntrinsics:
+    def test_init(self):
```

Comment:
It is necessary to add a test for initiating `CameraIntrinsics` with the argument `_init_distortion` given as `False`.